### PR TITLE
First step towards using bzlmod

### PR DESCRIPTION
**Pull request by hzeller**
First step towards using bzlmod

 * Move projects that are available already in https://registry.bazel.build/
    from from load_external to MODULE.bazel.
 * This is not complete: there are probably more steps to simplify after
     the first submit.
  * Switching to bzlmod broke making the compilation-db as the action hooks
       don't seem to work anymore ( dev_utils/make-compilation-db.sh ).
       The used com_grail_bazel_compdb is not maintained anymore, so this needs
       to be updated to something more modern (hedronvision?).
       Since this is not actively needed in daily development (and I am the only
       one really using it), postponed for later.
    
 Issues: #931
    
  Also, needed to make Python rules to work well with
  --incompatible_default_to_explicit_init_py (https://github.com/bazelbuild/bazel/issues/10076)

### CODE REVIEW

**Comment by hzeller** [comment]
Draft for now. It seems there is an issue with the Python interpreter not finding `google.api`. Details need to be figured out. Probably something to do with grpc and the way the Python toolchain is installed.

```
  File "xls/tools/codegen_main_test.runfiles/_main/xls/tools/codegen_main_test.py", line 27, in <module>
    from xls.codegen import module_signature_pb2
  File "xls/tools/codegen_main_test.runfiles/_main/xls/codegen/module_signature_pb2.py", line 25, in <module>
    from google.api import field_behavior_pb2 as google_dot_api_dot_field__behavior__pb2
ModuleNotFoundError: No module named 'google.api'
```

File: dependency_support/load_external.bzl [hunk 1]
```diff
@@ -38,60 +38,6 @@ def load_external_repositories():
...
-        name = "boringssl",
-        # Commit date: 2024-06-24
-        # Note for updating: we need to use a commit from the main-with-bazel branch.
-        strip_prefix = "boringssl-e6b03733628149a89a1d18b3ef8f39aa1055aba8",
-        sha256 = "006596f84d9cc142d9d6c48600cf6208f9d24426943b05e8bcda06e523f69dc8",
-        urls = ["https://github.com/google/boringssl/archive/e6b03733628149a89a1d18b3ef8f39aa1055aba8.tar.gz"],
-    )
-
     # Commit on 2023-02-09
     http_archive(
```
Comment (proppy) [inline dependency_support/load_external.bzl:42]: do we still need that? (I thought we removed the python bindings but maybe one of our dependency uses it?)

Comment (hzeller) [reply dependency_support/load_external.bzl:42]: Maybe not. Since I didn't know what it is for, I didn't touch it.

**Comment by hzeller** [comment]
To reproduce, this is a good target:

```
bazel test -c opt //xls/tools:codegen_main_test
```

File: dependency_support/load_external.bzl (hunk 1 above)
Comment (allight) [reply dependency_support/load_external.bzl:42]: I'm pretty sure everything using pybind is gone.

**Comment by hzeller** [comment]
Also adding @cdleary - can you test that it works in your compilation environment ?

File: MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+bazel_dep(name = "riegeli", version = "0.0.0-20240606-973b6f0", repo_name = "com_google_riegeli")
+bazel_dep(name = "re2", version = "2024-07-02", repo_name = "com_googlesource_code_re2")
+bazel_dep(name = "googletest", version = "1.15.2", repo_name = "com_google_googletest")
+bazel_dep(name = "nlohmann_json", version = "3.11.3", repo_name = "jsonhpp")
+bazel_dep(name = "boringssl", version = "0.0.0-20240530-2db0eb3")
+
+# Roughly these are all the targets we need
+#  find . -name BUILD | xargs sed 's/.*"\(@[^/"]*\).*/\1/p;d' | sort | uniq
+
+## Not yet available as bzlmod or other problems to be solved.
```
Comment (ericastor) [inline MODULE.bazel:40]: This list doesn't seem comprehensive. Maybe we can move this to comments on the individual items remaining in our WORKSPACE or load_external.bzl?

**Comment by cdleary** [comment]
> Also adding @cdleary - can you test that it works in your compilation environment ?

Will try today thanks

File: MODULE.bazel (hunk 2 above)
Comment (hzeller) [reply MODULE.bazel]: Done.
  * kept the script example to find dependencies in MODULE.bzlmod
  * Added TODO comments in WORKSPACE, {load,initialize}_external.bzl describing how the goal is to have them empty :)

File: dependency_support/load_external.bzl [hunk 3]
```diff
@@ -278,15 +174,6 @@ def load_external_repositories():
...
-        name = "rules_license",
-        urls = [
-            "https://github.com/bazelbuild/rules_license/releases/download/0.0.8/rules_license-0.0.8.tar.gz",
-        ],
-        sha256 = "241b06f3097fd186ff468832150d6cc142247dc42a32aaefb56d0099895fd229",
-    )
-
     # 2022-09-19
     http_archive(
         name = "com_grail_bazel_compdb",
```
[the remark this answers is not in the document]
Comment (hzeller) [reply dependency_support/load_external.bzl]: Done. And replaced `xls/dev_tools/make-compilation-db.sh` content with a _sorry, work in progress_ message.

**Comment by hzeller** [comment]
(ah, looks like I messed up some rebase. will fix)

**Comment by hzeller** [comment]
Since I discombobulated the merge/rebase dance here, new, fresh PR in #1570

**Pull request closed without merging**